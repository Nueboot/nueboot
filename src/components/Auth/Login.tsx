import React from 'react';
import styled from 'styled-components';

import { Dropdown, DropdownContent } from '../Styled/Dropdown';
import Text from '../Styled/Text';

const HoverText = styled(Text)`
  :hover {
    font-weight: 700;
  }
`;

const Login: React.SFC = () => (
  <Dropdown>
    <Text>Login ‚åµ</Text>
    <DropdownContent>
      <HoverText>Google</HoverText>
      <HoverText>Facebook</HoverText>
    </DropdownContent>
  </Dropdown>
);

export default Login;
