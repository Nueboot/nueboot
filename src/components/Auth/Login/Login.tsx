import React from 'react';
import styled from 'styled-components';

import { Text } from '../../Styled';
import { Dropdown, DropdownContent } from '../../Styled/Dropdown';

export interface LoginProps {
  facebookLogin(): void;
  googleLogin(): void;
}

const HoverText = styled(Text)`
  :hover {
    font-weight: 700;
  }
`;

const Login: React.SFC<LoginProps> = ({ googleLogin, facebookLogin }) => (
  <Dropdown tabIndex={0}>
    <Text>Login ‚åµ</Text>
    <DropdownContent>
      <HoverText tabIndex={0} onClick={googleLogin}>
        Google
      </HoverText>
      <HoverText tabIndex={0} onClick={facebookLogin}>
        Facebook
      </HoverText>
    </DropdownContent>
  </Dropdown>
);

export default Login;
