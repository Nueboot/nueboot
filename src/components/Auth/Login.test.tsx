import { shallow, ShallowWrapper } from 'enzyme';
import React from 'react';

import Login from './Login';

describe('Login', () => {
  let logoutComponent: ShallowWrapper;

  beforeEach(() => {
    logoutComponent = shallow(<Login />);
  });

  it('renders correctly', () => {
    expect(logoutComponent).toMatchSnapshot();
  });
});
