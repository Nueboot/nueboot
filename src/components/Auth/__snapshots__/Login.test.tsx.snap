// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login renders correctly 1`] = `
<styled.div>
  <styled.span>
    Login ‚åµ
  </styled.span>
  <styled.div>
    <Styled(styled.span)>
      Google
    </Styled(styled.span)>
    <Styled(styled.span)>
      Facebook
    </Styled(styled.span)>
  </styled.div>
</styled.div>
`;
